### PR TITLE
183393419 group map results

### PR DESCRIPTION
**Pull request by kbiala**
#### What's this PR do?
- Adds an option to toggle nodes visibility (validators should always be visible)
- Groups map points into larger clusters

#### How should this be manually tested?
- If you have some data centers locally, you can skip this part. If you don't, you need to generate some localization data first. Turn on sidekiq and run:
```
rails r script/data_centers_scripts/append_data_centers_geo_data.rb
rails runner "DataCenterStatsWorker.perform_async('mainnet')"
```
or test on staging
- Go to /validators-map page and review. Click on nodes visibility button, click on points etc. Most of DCs should be grouped by country, DCs in US should be grouped by state.

#### What are the relevant tickets?
- [URL to the PivotalTracker ticket](https://www.pivotaltracker.com/n/projects/2177859/stories/183393419)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

File: app/javascript/packs/validators/validators_map.vue [hunk 1]
```diff
@@ -16,32 +16,59 @@
...
+        <div class="btn-group btn-group-toggle switch-button" v-else>
+          <span class="btn btn-xs btn-secondary" v-on:click="set_nodes_visibility(true)">
+            <i class="fas fa-eye me-2"></i>nodes
+          </span>
+          <span class="btn btn-xs btn-secondary active">
+            <i class="fas fa-eye-slash"></i>
+          </span>
+        </div>
+
+        <div class="d-none mt-3">
```
Comment (psuchta) [inline app/javascript/packs/validators/validators_map.vue:53]: Is it for future purposes?

File: app/javascript/packs/validators/validators_map.vue [hunk 2]
```diff
@@ -83,21 +110,64 @@
...
       },
 
-      select_data_center: function(data_center) {
-        this.selected_data_center = data_center;
-      }
+      set_map_point_class: function(validators_count, nodes_count) {
+        let point_size = this.set_map_point_size(validators_count + nodes_count);
+        let point_color = this.set_map_point_color(validators_count, nodes_count);
+
+        return "map-point " + point_size + " " + point_color;
```
Comment (psuchta) [inline app/javascript/packs/validators/validators_map.vue:147]: How about?
```suggestion
        return `map-point ${point_size} ${point_color}`;
```

File: app/controllers/api/v1/data_centers_controller.rb [hunk 3]
```diff
@@ -4,37 +4,69 @@ module Api
   module V1
     class DataCentersController < BaseController
       def index_with_nodes
+        show_gossip_nodes = dc_params[:show_gossip_nodes] == "false" ? false : true
```
Comment (psuchta) [inline app/controllers/api/v1/data_centers_controller.rb:7]: Alternatively
```suggestion
        show_gossip_nodes = dc_params[:show_gossip_nodes].in?(["true", nil])
```

**Review by psuchta — APPROVED** [review]

File: app/controllers/validators_controller.rb [hunk 4]
```diff
@@ -44,11 +44,6 @@ def home_new
     @network = validators_params[:network]
   end
 
-  # GET /validators-map
```
Comment (psuchta) [inline app/controllers/validators_controller.rb:47]: We can also delete this method from routes

Comment (kbiala) [reply app/controllers/validators_controller.rb:47]: I thought I did that. Thanks!

File: app/javascript/packs/validators/validators_map.vue (hunk 1 above)
Comment (kbiala) [reply app/javascript/packs/validators/validators_map.vue:53]: Yup, will be used in the next story

**Pull request merged**